src/packages/v4/components/Create/hooks/useLoadInitialStateFromQuery.ts

```diff
@@ -0,0 +1,178 @@
+  switch (duration) {
+    case '':
+      fundingCyclesPageSelection = undefined
+      break
+    case '0':
+      fundingCyclesPageSelection = 'manual'
+      break
+    default:
+      fundingCyclesPageSelection = 'automated'
+  }
+
+  const distributionLimit = initialState.fundAccessConstraints[0]
+    ?.distributionLimit
+    ? parseWad(initialState.fundAccessConstraints[0]?.distributionLimit)
+    : undefined
+
+  let treasurySelection: TreasurySelection | undefined
+
+  if (distributionLimit === undefined) {
+    treasurySelection = undefined
+  } else if (distributionLimit.eq(MAX_DISTRIBUTION_LIMIT)) {
+    treasurySelection = 'unlimited'
+  } else if (distributionLimit.eq(0)) {
+    treasurySelection = 'zero'
+  } else {
+    treasurySelection = 'amount'
+  }
+
+  let projectTokensSelection: ProjectTokensSelection | undefined
+  const initialTokenData = {
+    weight: initialState.fundingCycleData.weight,
+    reservedRate: initialState.fundingCycleMetadata.reservedRate,
+    reservedTokensGroupedSplits: initialState.reservedTokensGroupedSplits,
+    discountRate: initialState.fundingCycleData.discountRate,
+    redemptionRate: initialState.fundingCycleMetadata.redemptionRate,
+    allowMinting: initialState.fundingCycleMetadata.allowMinting,
+  }
+  if (isEqual(initialTokenData, ReduxDefaultTokenSettings)) {
+    projectTokensSelection = 'default'
+  } else {
+    projectTokensSelection = 'custom'
+  }
+
+  const reconfigurationRuleSelection =
+    ballotStrategiesFn({}).find(s =>
+      isEqualAddress(s.address, initialState.fundingCycleData.ballot),
+    )?.id ?? 'threeDay'
+
+  let createFurthestPageReached: CreatePage = 'projectDetails'
+  if (fundingCyclesPageSelection) {
+    createFurthestPageReached = 'fundingCycles'
+  }
+  if (treasurySelection) {
+    createFurthestPageReached = 'payouts'
+  }
+  if (projectTokensSelection) {
+    createFurthestPageReached = 'projectToken'
+  }
+  if (reconfigurationRuleSelection) {
+    createFurthestPageReached = 'reconfigurationRules'
+  }
+  if (
+    fundingCyclesPageSelection &&
+    treasurySelection &&
+    projectTokensSelection &&
+    reconfigurationRuleSelection
+  ) {
+    createFurthestPageReached = 'reviewDeploy'
+  }
+
+  return {
+    fundingCyclesPageSelection,
+    treasurySelection,
+    fundingTargetSelection: undefined, // TODO: Remove
+    payoutsSelection: undefined, // TODO: Remove
+    projectTokensSelection,
+    reconfigurationRuleSelection,
+    createFurthestPageReached,
+    createSoftLockPageQueue: undefined, // Not supported, this feature is used only for fully fledged projects.
+  }
+}
+
+/**
+ * Load redux state from a URL query parameter.
+ */
+export function useLoadingInitialStateFromQuery() {
+  const dispatch = useDispatch()
+  const router = useRouter()
+  const { contracts } = useJBContractContext()
+
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    if (!router.isReady || !contracts.primaryNativeTerminal.data) return
```

Comment:
Terminal address loaded for network given here: https://github.com/jbx-protocol/juice-interface/pull/4444/files#r1735468140